lib/config/folderconfiguration.go

```diff
@@ -206,6 +207,15 @@ func (f *FolderConfiguration) prepare(myID protocol.DeviceID, existingDevices ma
 	if f.MarkerName == "" {
 		f.MarkerName = DefaultMarkerName
 	}
+	if f.MarkerName == "." {
+		l.Warnf(`Setting the folder marker name to "." is not allowed, reverting to default "%v"`, DefaultMarkerName)
+		f.MarkerName = DefaultMarkerName
+	}
+	markerPath := filepath.Join(f.Path, f.MarkerName)
+	if s := strings.TrimPrefix(markerPath, f.Path); s == markerPath || s == "" {
```

Comment:
What about something like `foo/..` or `./.` etc?
Don't you need a filepath.Clean or something?

Reply:
`filepath.Join` already cleanes the result. Added those as testcases.